File: Practica 4/FigurasGeometricasSobre/src/FigurasGeometricasSobre.java
public class FigurasGeometricasSobre {

    public double area(double radio) {
        return Math.PI * radio * radio;
    }
    
    public double area(int base, int altura) {
        return base * altura;
    }
    
    public double area(double base, double altura) {
        return (base * altura) / 2.0;
    }
    
    public double area(double baseMayor, double baseMenor, double altura) {
        return ((baseMayor + baseMenor) * altura) / 2.0;
    }
    
    public double area(double base, int apotema) {
        return (base * apotema) / 2.0;
    }

    public static void main(String[] args) {
    	FigurasGeometricasSobre f1 = new FigurasGeometricasSobre();
    	FigurasGeometricasSobre f2 = new FigurasGeometricasSobre();
    	FigurasGeometricasSobre f3 = new FigurasGeometricasSobre();
    	FigurasGeometricasSobre f4 = new FigurasGeometricasSobre();
    	FigurasGeometricasSobre f5 = new FigurasGeometricasSobre();
        
        System.out.println(" círculo: " + f1.area(7));
        System.out.println(" rectángulo: " + f2.area(6, 4));
        System.out.println(" triángulo rectángulo: " + f3.area(4.0, 2.0));
        System.out.println(" trapecio: " + f4.area(3, 8, 3));
        System.out.println(" pentágono: " + f5.area(9.0, 4.0));
    }
}

File: Practica 4/FigurasGeometricasSobre.py
import math
 # sobrecarga

class Figura:
    def area(self):
        pass 
class Circulo(Figura):
    def __init__(self, radio):
        self.radio = radio

    def area(self):
        return math.pi * self.radio ** 2

class Rectangulo(Figura):
    def __init__(self, base, altura):
        self.base = base
        self.altura = altura

    def area(self):
        return self.base * self.altura

class TrianguloRect(Figura):
    def __init__(self, base, altura):
        self.base = base
        self.altura = altura

    def area(self):
        return (self.base * self.altura) / 2

class Trapecio(Figura):
    def __init__(self, base_mayor, base_menor, altura):
        self.base_mayor = base_mayor
        self.base_menor = base_menor
        self.altura = altura

    def area(self):
        return ((self.base_mayor + self.base_menor) * self.altura) / 2

class Pentagono(Figura):
    def __init__(self, perimetro, apotema):
        self.perimetro = perimetro
        self.apotema = apotema

    def area(self):
        return (self.perimetro * self.apotema) / 2

if __name__ == "__main__":
    f1 = Circulo(7)
    f2 = Rectangulo(6, 4)
    f3= TrianguloRect(4, 2)
    f4 = Trapecio(3, 8, 3)
    f5= Pentagono(9, 4)

    print("Círculo:", f1.area())
    print("Rectángulo:", f2.area())
    print("Triángulo rectángulo:", f3.area())
    print("Trapecio:", f4.area())
    print("Pentágono:", f5.area())
